tangelo/problem_decomposition/dmet/dmet_problem_decomposition.py

```diff
@@ -569,6 +601,86 @@ def _compute_energy(self, mf_frag, onerdm, twordm, fock_frag_copy, t_list, onein
+            cc_onerdm (numpy.array): one-particle reduced density matrix (float64).
+            cc_twordm (numpy.array): two-particle reduced density matrix (float64).
+            fock_frag_copy (numpy.array): Fock matrix with the chemical potential subtracted (float64).
+            t_list (list): List of number of fragment and bath orbitals (int).
+            oneint (numpy.array): One-electron integrals of fragment (float64).
+            twoint (numpy.array): Two-electron integrals of fragment (float64).
+            fock (numpy.array): Fock matrix of fragment (float64).
+
+        Returns:
+            float64: Fragment energy (fragment_energy).
+            float64: Total energy for fragment using RDMs (total_energy_rdm).
+            numpy.array: One-particle RDM for a fragment (one_rdm, float64).
+        """
+
+        # Execute CCSD calculation
+        norb = t_list[0]
+
+        # Calculate the one- and two- RDM for DMET energy calculation (Transform to AO basis)
+        one_rdm_a, one_rdm_b = onerdm
+
+        onerdm_a = reduce(np.dot, (mf_frag.mo_coeff[0], one_rdm_a, mf_frag.mo_coeff[0].T))
+        onerdm_b = reduce(np.dot, (mf_frag.mo_coeff[1], one_rdm_b, mf_frag.mo_coeff[1].T))
+
+        two_rdm_aa, two_rdm_ab, two_rdm_bb = twordm
+
+        twordm_aa = np.einsum("pi,ijkl->pjkl", mf_frag.mo_coeff[0], two_rdm_aa)
+        twordm_aa = np.einsum("qj,pjkl->pqkl", mf_frag.mo_coeff[0], twordm_aa)
+        twordm_aa = np.einsum("rk,pqkl->pqrl", mf_frag.mo_coeff[0], twordm_aa)
+        twordm_aa = np.einsum("sl,pqrl->pqrs", mf_frag.mo_coeff[0], twordm_aa)
+
+        twordm_ab = np.einsum("pi,ijkl->pjkl", mf_frag.mo_coeff[0], two_rdm_ab)
+        twordm_ab = np.einsum("qj,pjkl->pqkl", mf_frag.mo_coeff[0], twordm_ab)
+        twordm_ab = np.einsum("rk,pqkl->pqrl", mf_frag.mo_coeff[1], twordm_ab)
+        twordm_ab = np.einsum("sl,pqrl->pqrs", mf_frag.mo_coeff[1], twordm_ab)
+
+        twordm_bb = np.einsum("pi,ijkl->pjkl", mf_frag.mo_coeff[1], two_rdm_bb)
+        twordm_bb = np.einsum("qj,pjkl->pqkl", mf_frag.mo_coeff[1], twordm_bb)
+        twordm_bb = np.einsum("rk,pqkl->pqrl", mf_frag.mo_coeff[1], twordm_bb)
+        twordm_bb = np.einsum("sl,pqrl->pqrs", mf_frag.mo_coeff[1], twordm_bb)
+
+        # Calculate the total energy based on RDMs
+        total_energy_rdm = np.einsum("ij,ij->", fock_frag_copy, onerdm_a ) \
+                         + np.einsum("ij,ij->", fock_frag_copy, onerdm_b ) \
+                         + 0.5 * np.einsum("ijkl,ijkl->", twoint, twordm_aa ) \
+                         + 0.5 * np.einsum("ijkl,ijkl->", twoint, twordm_ab ) \
+                         + 0.5 * np.einsum("ijkl,klij->", twoint, twordm_ab ) \
+                         + 0.5 * np.einsum("ijkl,ijkl->", twoint, twordm_bb )
+
+        # Calculate fragment expectation value
+        fragment_energy_one_rdm = 0.25 * np.einsum("ij,ij->", onerdm_a[: norb, :], fock[: norb, :] + oneint[: norb, :]) \
```

Comment:
Remove 0.125 and 0.25 from all these lines, simply rescale the result of these big lines after we summed all the einsums.